src/browser/api_protocol/api_handlers/webcontents.ts

```diff
@@ -0,0 +1,130 @@
+import { Identity, APIMessage, Acker, APIPayloadAck } from '../../../shapes';
+
+import { getTargetWindowIdentity, registerActionMap } from './api_protocol_base';
+
+import * as WebContents from '../../api/webcontents';
+import { getWindowByUuidName } from '../../core_state';
+const { Application } = require('../../api/application');
+const successAck: APIPayloadAck = { success: true };
+
+export const webContentsApiMap = {
+    'execute-javascript-in-window': { apiFunc: executeJavascript, apiPath: '.executeJavaScript' },
+    'get-zoom-level': getZoomLevel,
+    'navigate-window': navigateWindow,
+    'navigate-window-back': navigateWindowBack,
+    'navigate-window-forward': navigateWindowForward,
+    'stop-window-navigation': stopWindowNavigation,
+    'reload-window': reloadWindow,
+    'set-zoom-level': setZoomLevel
+};
+export function init () {
+    registerActionMap(webContentsApiMap, 'Window');
+}
+function executeJavascript(identity: Identity, message: APIMessage, ack: Acker, nack: (error: Error) => void): void {
+    const { payload } = message;
+    const { code } = payload;
+    const dataAck = Object.assign({}, successAck);
+    const windowIdentity = getTargetWindowIdentity(payload);
+    const browserWin = getElectronBrowserWindow(windowIdentity);
+
+    let { uuid: pUuid } = windowIdentity;
+
+    while (pUuid) {
+        if (pUuid === identity.uuid) {
+            return WebContents.executeJavascript(browserWin.webContents, code, (err: Error, result: any) => {
+                if (err) {
+                    nack(err); // TODO: this nack doesn't follow the protocol
+                } else {
+                    dataAck.data = result;
+                    ack(dataAck);
+                }
+            });
+        }
+        pUuid = Application.getParentApplication({
+            uuid: pUuid
+        });
+    }
+
+    return nack(new Error('Rejected, target window is not owned by requesting identity'));
+}
+function navigateWindow(identity: Identity, message: APIMessage, ack: Acker, nack: (error: Error) => void): void {
```

Comment:
Does the name still make sense after moving out of window?

Reply:
(this applies to other methods here too)

Reply:
I'd rather rename in a follow up pr, for now it still only does windows and I'm trying to keep changes to a minimum

Reply:
No rename is needed, the API action cannot change `navigate-window` and the only purpose of the API name is to be mapped.

Reply:
@rdepena I assume @tomer-openfin was referencing the function names, not the api mapping

Reply:
right, but the function is only used by the mapping so it should just keep the name.